prefect_docker/images.py

```diff
@@ -0,0 +1,84 @@
+"""Integrations with Docker Images."""
+
+from typing import List, Optional, Union
+
+from docker.models.images import Image
+from prefect import get_run_logger, task
+from prefect.utilities.asyncutils import run_sync_in_worker_thread
+
+from prefect_docker.credentials import DockerRegistryCredentials
+from prefect_docker.host import DockerHost
+
+
+@task
+async def pull_docker_image(
+    repository: str,
+    tag: Optional[str] = None,
+    platform: Optional[str] = None,
+    all_tags: Optional[bool] = None,
+    docker_host: Optional[DockerHost] = None,
+    docker_registry_credentials: Optional[DockerRegistryCredentials] = None,
+    **pull_kwargs,
+) -> Union[Image, List[Image]]:
```

Comment:
This type reflects the code better, but I wonder if it would be more useful to return the who `Image` object instead of just the ID.
```suggestion
) -> Union[str, List[str]]:
```

Reply:
I initially thought it would cause serialization issues, but it worked so I changed it back to image.